### PR TITLE
Bug fixes for service of notice - 77218/77219

### PR DESCRIPTION
**Pull request by will-hope98**

### CODE REVIEW

File: src/EA.Weee.Web/Areas/Producer/Controllers/ProducerSubmissionController.cs [hunk 1]
```diff
@@ -108,7 +108,10 @@ public async Task<ActionResult> ServiceOfNotice(bool? sameAsOrganisationAddress)
         {
             var model = mapper.Map<SmallProducerSubmissionData, ServiceOfNoticeViewModel>(SmallProducerSubmissionData);
 
-            model.Address = new ServiceOfNoticeAddressData();
+            if (model.Address == null)
+            {
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Producer/Controllers/ProducerSubmissionController.cs:112]: @will-hope98  If this change resolved a bug, really we should create a unit test to cover it. To keep the action as clean as possible I would move this into the mapper.

File: src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/ServiceOfNoticeMap.cs [hunk 2]
```diff
@@ -23,6 +23,7 @@ public ServiceOfNoticeViewModel Map(SmallProducerSubmissionData source)
                 DirectRegistrantId = source.DirectRegistrantId,
                 OrganisationId = source.OrganisationData.Id,
                 Address = address,
+                HasAuthorisedRepresentitive = source.HasAuthorisedRepresentitive,
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/ServiceOfNoticeMap.cs:26]: @will-hope98  Update the unit test.

File: src/EA.Weee.Web/Areas/Producer/Controllers/ProducerSubmissionController.cs (hunk 1 above)
Comment (will-hope98) [reply src/EA.Weee.Web/Areas/Producer/Controllers/ProducerSubmissionController.cs]: Done

File: src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/ServiceOfNoticeMap.cs (hunk 2 above)
Comment (will-hope98) [reply src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/ServiceOfNoticeMap.cs:26]: The unit test already has this

**Pull request merged**